### PR TITLE
Range projectiles that shoot out on right click

### PR DESCRIPTION
**Pull request by OtakuFreak13**
These projectiles will apply damage if the collided entity has a health component.

### CODE REVIEW

File: StortSpelProjekt/Engine/src/Renderer/Transform.cpp [hunk 1]
```diff
@@ -190,6 +190,20 @@ DirectX::XMMATRIX Transform::GetRotMatrix() const
...
+	DirectX::XMStoreFloat3(&forward, m_RotationMat.r[2]);
+	return forward;
+}
+
+DirectX::XMFLOAT3 Transform::GetUpVec() const
+{
+	DirectX::XMFLOAT3 up;
+	DirectX::XMStoreFloat3(&up, m_RotationMat.r[1]);
+	return up;
+}
```
Comment (kattalainen) [inline StortSpelProjekt/Engine/src/Renderer/Transform.cpp]: These functions have been added in develop from another branch, make sure there are not duplicates when you merge

File: StortSpelProjekt/Engine/src/Renderer/Transform.h [hunk 2]
```diff
@@ -46,6 +46,8 @@ class Transform
 	DirectX::XMFLOAT3 GetScale() const;
 	// gets the rotation of the transform in all axisis
 	DirectX::XMMATRIX GetRotMatrix() const;
+	DirectX::XMFLOAT3 GetForwardVec() const;
+	DirectX::XMFLOAT3 GetUpVec() const;
```
Comment (kattalainen) [inline StortSpelProjekt/Engine/src/Renderer/Transform.h]: These functions have been added in develop from another branch, make sure there are not duplicates when you merge

File: StortSpelProjekt/Game/src/Gamefiles/Components/HealthComponent.cpp [hunk 3]
```diff
@@ -44,19 +42,15 @@ int component::HealthComponent::GetHealth()
...
 
 void component::HealthComponent::printDeath(Death* event)
 {
 	if (event->ent == m_pParent)
 	{
 		Log::Print("%s died!\n", event->ent->GetName().c_str());
+		//if (m_pParent->HasComponent<component::AiComponent>())
+		//{
+		//	m_pParent->GetComponent<component::AiComponent>()
+		//}
```
Comment (kattalainen) [inline StortSpelProjekt/Game/src/Gamefiles/Components/HealthComponent.cpp]: remove unnecessary comments

File: StortSpelProjekt/Game/src/Gamefiles/Components/ProjectileComponent.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+component::ProjectileComponent::~ProjectileComponent()
+{
+
+}
+
+void component::ProjectileComponent::Update(float dt)
+{
+	//m_pParent->GetComponent<component::TransformComponent>()->GetTransform()->Move(dt);
```
Comment (kattalainen) [inline StortSpelProjekt/Game/src/Gamefiles/Components/ProjectileComponent.cpp]: remove unnecessary comments

**Review by kattalainen — DISMISSED** [review]

**Review by Whixx — APPROVED** [review]
Seems to work fine

**Review by kattalainen — APPROVED** [review]

**Pull request merged**